components/enrichers/custom-annotation/main_test.go

```diff
@@ -0,0 +1,139 @@
+	require.NoError(t, run())
+
+	// Check there is something in our output directory
+	files, err := os.ReadDir(outdir)
+	require.NoError(t, err)
+	require.NotEmpty(t, files)
+	require.Len(t, files, 4)
+
+	// Check that both of them are EnrichedLaunchToolResponse
+	// and their Issue property is an empty list
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), ".raw.pb") {
+			continue
+		}
+
+		encodedProto, err := os.ReadFile(fmt.Sprintf("%s/%s", outdir, f.Name()))
+		require.NoError(t, err)
+		output := &draconv1.EnrichedLaunchToolResponse{}
+		require.NoError(t, proto.Unmarshal(encodedProto, output))
+		require.Empty(t, output.Issues)
+	}
+}
+
+func TestHandlesFindings(t *testing.T) {
+	indir, outdir := enrichers.SetupIODirs(t)
+	annotation = "This is my custom annotation"
+	annotationValue = "This is my custom annotation value"
+
+	mockLaunchToolResponses := enrichers.GetLaunchToolResponse(t)
+	for i, r := range mockLaunchToolResponses {
+		// Write sample enriched responses to indir
+		encodedProto, err := proto.Marshal(r)
+		require.NoError(t, err)
+		rwPermission600 := os.FileMode(0o600)
+		require.NoError(t, os.WriteFile(fmt.Sprintf("%s/input_%d_%s.tagged.pb", indir, i, r.ToolName), encodedProto, rwPermission600))
+	}
+
+	// Run the enricher
+	enrichers.SetReadPathForTests(indir)
+	enrichers.SetWritePathForTests(outdir)
+	require.NoError(t, run())
+
+	// Check there is something in our output directory
+	files, err := os.ReadDir(outdir)
+	require.NoError(t, err)
+	require.NotEmpty(t, files)
+	require.Len(t, files, 4)
+
+	// Check that both of them are EnrichedLaunchToolResponse
+	// and their Issue property is not an empty list
+	expected := []*draconv1.EnrichedLaunchToolResponse{
+		{
+			OriginalResults: mockLaunchToolResponses[0],
+			Issues: []*draconv1.EnrichedIssue{
+				{
+					RawIssue: mockLaunchToolResponses[0].Issues[0],
+					Annotations: map[string]string{
+						annotation: annotationValue,
+					},
+				},
+				{
+					RawIssue: mockLaunchToolResponses[0].Issues[1],
+					Annotations: map[string]string{
+						annotation: annotationValue,
+					},
+				},
+			},
+		},
+		{
+			OriginalResults: mockLaunchToolResponses[1],
+			Issues: []*draconv1.EnrichedIssue{
+				{
+					RawIssue: mockLaunchToolResponses[1].Issues[0],
+					Annotations: map[string]string{
+						annotation: annotationValue,
+					},
+				},
+				{
+					RawIssue: mockLaunchToolResponses[1].Issues[1],
+					Annotations: map[string]string{
+						annotation: annotationValue,
+					},
+				},
+			},
+		},
+	}
+	var actual draconv1.EnrichedLaunchToolResponse
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), ".raw.pb") {
+			continue
+		}
+		encodedProto, err := os.ReadFile(fmt.Sprintf("%s/%s", outdir, f.Name()))
+		require.NoError(t, err)
+
+		require.NoError(t, proto.Unmarshal(encodedProto, &actual))
+		if actual.OriginalResults.ToolName == expected[0].OriginalResults.ToolName {
+			if !proto.Equal(&actual, expected[0]) {
```

Comment:
this could be simplified to `require.True(t, proto.Equal(&actual, expected[0]), cmp.Diff(&actual, expected[0], protocmp.Transform()))`